### ISSUE TITLE
Blank screen capture on iOS & Cordova

### ISSUE THREAD

**Issue by sebastianzillessen** [niklasvh/html2canvas#333]
I use your plugin in an Cordova 3.1 application on iOS.

Unfortunately the canvas stays empty. 

If I print the base64 encoded version of this canvas via `var data = canvas.toDataURL('image/png');` I get the logs below:

```
 html2canvas: Preload starts: finding background-images
 html2canvas: Preload: Finding images
 html2canvas: Preload: Done.
 html2canvas: start: images: 1 / 2 (failed: 0)
 html2canvas: start: images: 2 / 2 (failed: 0)
 Finished loading images: # 2 (failed: 0)
 html2canvas: Renderer: Canvas renderer done - returning canvas obj

Canvas returned successfully
data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAUAAAAHMCAYAAACk4nEJAAAKIElEQVR4Ae3QgQAAAADDoPlTH+SFUGHAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDBgwYMGDAgAEDB ….

```

Has anyone tried to use this on iOS, probably even with Cordova / Phonegap?

Thanks!

**Comment by sjamaan** [comment]
I also had this problem.  It seems that the offset calculations are wrong, causing it to render "off screen" content outside the canvas.

I don't know how to fix it properly, but as a workaround, you can force the browser to scroll to the top of the screen before making the screenshot. Make sure you make the screenshot after the scroll has completed, to leave the browser time to actually repaint before capturing.

```javascript
                   var maybeScrollAndMakeScreenshot = function() {
                        if (window.scrollY > 0) {
                            // Ugly workaround for bug in M&%^$@%# Android webview
                            $(window.document).one('scroll', makeScreenshot);
                            window.scroll(0,0);
                        } else {
                            makeScreenshot();
                        }
                    };
```

The makeScreenshot function just calls html2canvas and saves the image data.

**Comment by usmonster** [comment]
Maybe fixed in 98ee306? Please test with the latest code on the master branch.

**Comment by Abhi-joshi** [comment]
Any workaround for phonegap.. Still seeing a cross default image.
